bokeh/plotting/helpers.py

```diff
@@ -68,6 +67,110 @@ def _stack(stackers, spec0, spec1, **kw):
+                err=err.message
+            )
+            reraise(ValueError, ValueError(msg), sys.exc_info()[2])
+
+    if not isinstance(edge_source, ColumnarDataSource):
+        try:
+            # try converting the soruce to ColumnDataSource
+            edge_source = ColumnDataSource(edge_source)
+        except ValueError as err:
+            msg = "Failed to auto-convert {curr_type} to ColumnDataSource.\n Original error: {err}".format(
+                curr_type=str(type(edge_source)),
+                err=err.message
+            )
+            reraise(ValueError, ValueError(msg), sys.exc_info()[2])
+
+    ## node stuff
+    if any(x.startswith('node_selection_') for x in kwargs):
+        snode_ca = _pop_colors_and_alpha(Circle, kwargs, prefix="node_selection_")
+    else:
+        snode_ca = None
+
+    if any(x.startswith('node_hover_') for x in kwargs):
+        hnode_ca = _pop_colors_and_alpha(Circle, kwargs, prefix="node_hover_")
+    else:
+        hnode_ca = None
+
+    if any(x.startswith('node_muted_') for x in kwargs):
+        mnode_ca = _pop_colors_and_alpha(Circle, kwargs, prefix="node_muted_")
+    else:
+        mnode_ca = None
+
+    nsnode_ca = _pop_colors_and_alpha(Circle, kwargs, prefix="node_nonselection_")
+    node_ca = _pop_colors_and_alpha(Circle, kwargs, prefix="node_")
+
+    ## edge stuff
+    if any(x.startswith('edge_selection_') for x in kwargs):
+        sedge_ca = _pop_colors_and_alpha(MultiLine, kwargs, prefix="edge_selection_")
+    else:
+        sedge_ca = None
+
+    if any(x.startswith('edge_hover_') for x in kwargs):
+        hedge_ca = _pop_colors_and_alpha(MultiLine, kwargs, prefix="edge_hover_")
+    else:
+        hedge_ca = None
+
+    if any(x.startswith('edge_muted_') for x in kwargs):
+        medge_ca = _pop_colors_and_alpha(MultiLine, kwargs, prefix="edge_muted_")
+    else:
+        medge_ca = None
+
+    nsedge_ca = _pop_colors_and_alpha(MultiLine, kwargs, prefix="edge_nonselection_")
+    edge_ca = _pop_colors_and_alpha(MultiLine, kwargs, prefix="edge_")
+
+    ## node stuff
+    node_kwargs = {k: v for k, v in kwargs.copy().items() if k in Circle.properties()}
+
+    node_glyph = _make_glyph(Circle, node_kwargs, node_ca)
+    nsnode_glyph = _make_glyph(Circle, node_kwargs, nsnode_ca)
+    snode_glyph = _make_glyph(Circle, node_kwargs, snode_ca)
+    hnode_glyph = _make_glyph(Circle, node_kwargs, hnode_ca)
+    mnode_glyph = _make_glyph(Circle, node_kwargs, mnode_ca)
+
+    node_renderer = GlyphRenderer(glyph=node_glyph,
+                                  nonselection_glyph=nsnode_glyph,
+                                  selection_glyph=snode_glyph,
+                                  hover_glyph=hnode_glyph,
+                                  muted_glyph=mnode_glyph,
+                                  data_source=node_source)
+
+    ## edge stuff
+    edge_kwargs = {k: v for k, v in kwargs.copy().items() if k in MultiLine.properties()}
```

Comment:
I'm unsure what the best route is here. Should all of the Glyph props be prefixed with `edge_` and `node_`?

The issue i'm running into is if a user does:
```python
plot.graph(..., line_width=10)
```
The line_width kwarg is applied to be the node (Circle) glyph and the edge (MultiLine) glyph.

Reply:
> Should all of the Glyph props be prefixed with edge_ and node_?

I think that makes sense. Similar to e.g. `nonselection_line_color`